dotCMS/src/main/java/com/dotcms/concurrent/DotConcurrentFactory.java

```diff
@@ -342,6 +379,156 @@ public IdentifierStripedLock getIdentifierStripedLock(){
        return this.identifierStripedLock;
     }
 
+    /**
+     * {@link SubmitterConfigCreator} builder
+     */
+    public static class SubmitterConfigCreatorBuilder {
+
+        private ThreadFactory            threadFactory;
+        private RejectedExecutionHandler rejectedExecutionHandler;
+        private Boolean allowCoreThreadTimeOut;
```

Comment:
Codacy found an issue: [Field allowCoreThreadTimeOut has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6265363)

Reply:
why not use the primitives and avoid a possible npe?

Reply:
actually the idea is to have objects instead of primitives, if null uses the fallback implementation, so only the non-null properties will overrides the default ones.